docs/rules/no-unnecessary-act.md

```diff
@@ -0,0 +1,26 @@
+# Disallow the use of `act` when wrapping Testing Library functions or functions with an empty body
+
+## Rule Details
+
+This rule disallows the usage of `act` when using it to wrap Testing Library functions, or functions with an empty body, which suppresses valid warnings. For more information, see [https://kcd.im/react-act](https://kcd.im/react-act).
+
+Examples of **incorrect** code for this rule:
+
+```js
+import { fireEvent, act } from '@testing-library/react';
```

Comment:
I would actually use both examples: with `fireEvent` to clarify it's a Testing Library util already wrapped with `act`, and `userEvent` to clarify the same since it's imported from a different package.